Quota configuration documentation update (#110056)
openshift/openshift-docs: modules/ols-activating-token-quota-limits.adoc

[role="_abstract"]

To manage resource consumption, activate token quota limits for {ols-long} by adding the `quotaHandlersConfig` section to the `OLSConfig` custom resource (CR). When you update the CR, the {ols-long} Operator automatically generates the required internal configuration for the service.

.Prerequisites

* You have installed the {ols-long} Operator.

* You have configured a large language model (LLM) provider.

* The {ols-long} service has access to a PostgreSQL database.

.Procedure

> . Edit the {ols-long} `OLSconfig` CR file:

Reviewer: ```suggestion
. Edit the {ols-long} `OLSconfig` CR file by running the following command:
```
This is not incorrect. Almost every OpenShift writer uses this style, so let's not change it.

+
[source,terminal]
----
$ oc edit olsconfig cluster
----

> . Update the `spec.ols.quotaHandlersConfig` section to include your token quota limit configuration.

Reviewer: ```suggestion
. Update the `spec.ols.quotaHandlersConfig` specification to include the token quota limit configuration.
```
This is field inside specification, so specification is correct. Let me know if you have any questions. :)

spec:
  ols:
    quotaHandlersConfig:
      enableTokenHistory: true
      limitersConfig:
      - name: user_monthly_limits
        type: user_limiter
        initialQuota: 100000
        quotaIncrease: 1000
        period: 30 days
      - name: cluster_monthly_limits
        type: cluster_limiter 
        initialQuota: 1000000
        quotaIncrease: 100000
        period: 30 days
----

> ** Configure the following fields based on your requirements:

Reviewer: I have a suggestion to reform/revert this change based on our [style guide](https://redhat-documentation.github.io/supplementary-style-guide/#explain-commands-variables-in-code-blocks). I will add the suggestion in line. Do refer to the style guide to understand why I made these changes or you can reach out to me.

```suggestion
```

+
--

> *General configuration*

Reviewer: ```suggestion
* General configuration:
```

> `enableTokenHistory`:: Enables the tracking of token usage history in the PostgreSQL database.

Reviewer: ```suggestion
** `spec.ols.quotaHandlersConfig.enableTokenHistory` enables the tracking of token usage history in the PostgreSQL database.
```

> `period`:: Defines the duration before the quota resets or the quota limit increases.

Reviewer: ```suggestion
** ``spec.ols.quotaHandlersConfig.limitersConfig.period` defines the duration before the quota resets or the quota limit increases.
```

> *User quota configuration (`user_limiter`)*

Reviewer: ```suggestion
* User quota configuration (`user_limiter`):
```

`name`:: Specifies a unique name for the user quota limiter.
`initialQuota`:: Sets the initial number of tokens available for each user (for example, `100000`).

> `quotaIncrease`:: Specifies the number of tokens added to the quota at the end of each period.

Reviewer: ```suggestion
** `spec.ols.quotaHandlersConfig.limitersConfig.name` specifies a unique name for the user quota limiter.
** `spec.ols.quotaHandlersConfig.limitersConfig.initialQuota` sets the initial number of tokens available for each user (for example, `100000`).
** `spec.ols.quotaHandlersConfig.limitersConfig.quotaIncrease` specifies the number of tokens added to the quota at the end of each period.
```

> *Cluster quota configuration (`cluster_limiter`)*

Reviewer: ```suggestion
* Cluster quota configuration (`cluster_limiter`):
```

 `name`:: Specifies a unique name for the cluster quota limiter.
 `initialQuota`:: Sets the initial number of tokens available for the entire cluster (for example, `100000`).

>  `quotaIncrease`:: Specifies the number of tokens added to the cluster quota at the end of each period.

Reviewer: ```suggestion
** `spec.ols.quotaHandlersConfig.limitersConfig.name` specifies a unique name for the cluster quota limiter.
** `spec.ols.quotaHandlersConfig.limitersConfig.initialQuota` sets the initial number of tokens available for the entire cluster (for example, `100000`).
** `spec.ols.quotaHandlersConfig.limitersConfig.quotaIncrease` specifies the number of tokens added to the cluster quota at the end of each period.
```

--

. Save the changes and exit the editor. 

.Verification

> * Confirm that the configuration is active by checking the {ols-long} Operator logs:

Reviewer: ```suggestion
* Confirm that the configuration is active by checking the {ols-long} Operator logs by running the following command:
```

+
[source,terminal]
----
$ oc logs -l app.kubernetes.io/name=lightspeed-operator -n openshift-lightspeed
----

> * Alternatively, verify the CR was saved correctly.

Reviewer: ```suggestion
* Alternatively, verify that the CR was saved correctly by running the following command:
```
